src/mngt/ocf_mngt_cache.c

```diff
@@ -1562,6 +1583,79 @@ struct ocf_pipeline_properties _ocf_mngt_cache_load_pipeline_properties = {
 	},
 };
 
+static void _ocf_mngt_init_cache_front_volume(ocf_pipeline_t pipeline,
+		void *priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_cache_attach_context *context = priv;
+	ocf_cache_t cache = context->cache;
+	ocf_volume_type_t type;
+	struct ocf_volume_uuid uuid = {
+		.data = cache,
+		.size = sizeof(cache),
+	};
+	int result;
+
+	type = ocf_ctx_get_volume_type_internal(cache->owner, OCF_VOLUME_TYPE_CACHE);
+	if (!type)
+		OCF_PL_FINISH_RET(context->pipeline, -OCF_ERR_INVAL);
+
+	result = ocf_volume_init(&cache->device->front_volume, type, &uuid, false);
+	if (result)
+		OCF_PL_FINISH_RET(context->pipeline, result);
+	context->flags.front_volume_inited = true;
+
+	result = ocf_volume_open(&cache->device->front_volume, NULL);
+	if (result)
+		OCF_PL_FINISH_RET(context->pipeline, result);
+	context->flags.front_volume_opened = true;
+
+	ocf_pipeline_next(context->pipeline);
+}
+
+static void _ocf_mngt_load_unsafe_complete(void *priv, int error)
+{
+	struct ocf_cache_attach_context *context = priv;
+	ocf_cache_t cache = context->cache;
+
+	if (error) {
+		ocf_cache_log(cache, log_err,
+				"ERROR: Cannot load metadata\n");
+		OCF_PL_FINISH_RET(context->pipeline,
+				-OCF_ERR_START_CACHE_FAIL);
+	}
+
+	ocf_pipeline_next(context->pipeline);
+}
+
+static void _ocf_mngt_load_metadata_unsafe(ocf_pipeline_t pipeline,
+		void *priv, ocf_pipeline_arg_t arg)
+{
+	struct ocf_cache_attach_context *context = priv;
+	ocf_cache_t cache = context->cache;
+
+	ocf_cache_log(cache, log_info, "Loading cache state...\n");
+	ocf_metadata_load_unsafe(cache,
+			_ocf_mngt_load_unsafe_complete, context);
+}
+
+struct ocf_pipeline_properties _ocf_mngt_cache_bind_pipeline_properties = {
+	.priv_size = sizeof(struct ocf_cache_attach_context),
+	.finish = _ocf_mngt_cache_attach_finish,
+	.steps = {
+		OCF_PL_STEP(_ocf_mngt_copy_uuid_data),
+		OCF_PL_STEP(_ocf_mngt_init_attached_nonpersistent),
+		OCF_PL_STEP(_ocf_mngt_attach_cache_device),
+		OCF_PL_STEP(_ocf_mngt_init_cache_front_volume),
+		OCF_PL_STEP(_ocf_mngt_init_properties),
+		OCF_PL_STEP(_ocf_mngt_attach_check_ram),
+		OCF_PL_STEP(_ocf_mngt_test_volume),
+		OCF_PL_STEP(_ocf_mngt_init_properties),
```

Comment:
Is `_ocf_mngt_init_properties()` called twice on purpose?

Reply:
No. Fixed.